src/com/sproutlife/model/GameModel.java

```diff
@@ -15,102 +15,190 @@
-        this.settings = settings;
-        this.clock = new GameClock();
-        echosystem = new Echosystem(clock);
-        gameStep = new GameStep(this);       
-        gameThread = new GameThread(this, interactionLock);
-        stats = new Stats(this);
-    }
-
-    public void performGameStep() {
-        incrementTime();
-        gameStep.perform();
-    }
-    /*
-    public GameStep getGameStep() {
-        return gameStep;
-    }
-    */
-    public Echosystem getEchosystem() {
-        return echosystem;
-    }
-
-    public Board getBoard() {
-        return echosystem.getBoard();
-    }          
-
-    public int getTime() {
-        return clock.getTime();
-    }
-
-    public GameClock getClock() {
-        return clock;
-    }
-
-    private void incrementTime() {
-        clock.increment();
-    }
-
-    public void resetGame() {
-        getEchosystem().resetCells();        
-        getEchosystem().pruneEmptyOrganisms();
-        getEchosystem().clearRetiredOrgs();
-        getStats().reset();
-        getClock().reset();
-        
-    }
-    
-    public Stats getStats() {
-        return stats;
-    }   
-
-    public Settings getSettings() {
-        return settings;
-    }
-    
-
-    public void setPlayGame(boolean playGame) {
-        if (playGame) {          
-            gameThread.setPlayGame(true);
-
-        } else {
-            gameThread.setPlayGame(false);            
-        }
-    }
-    
-    public GameThread getGameThread() {
-        return gameThread;
-    }
-    
-    public boolean getPlayGame() {
-        return gameThread.getPlayGame();
-    }
-
-    public void setGameStepListener(GameStepListener l) {
-        if (gameThread!=null) {
-            gameThread.setGameStepListener(l);
-            gameStep.setGameStepListener(l);
-        }
-    }
-
-    public void set(String s, Object o) {
-        getSettings().set(s,o);
-    }
+	/**
+	 * Calls Echosystem default constructor.
+	 */
+	private Echosystem echosystem;
+
+	/**
+	 * Calls GameClock default constructor.
+	 */
+	private GameClock clock;
+
+	/**
+	 * Calls GameStep default constructor.
+	 */
+	private GameStep gameStep;
+
+	/**
+	 * Calls GameThread default constructor.
+	 */
+	private GameThread gameThread;
+
+	/**
+	 * Calls Settings default constructor.
+	 */
+	private Settings settings;
+
+	/**
+	 * Calls Stats default constructor.
+	 */
+	private Stats stats;
+
+	/**
+	 * GameModel Constructor
+	 */
+	public GameModel(Settings settings, ReentrantReadWriteLock interactionLock) {
+		this.settings = settings;
+		this.clock = new GameClock();
+		echosystem = new Echosystem(clock);
+		gameStep = new GameStep(this);
+		gameThread = new GameThread(this, interactionLock);
+		stats = new Stats(this);
+	}
+
+	/**
+	 * Performs a single game step.
+	 */
+	public void performGameStep() {
+		incrementTime();
+		gameStep.perform();
+	}
+
+	/*
+	 * public GameStep getGameStep() { return gameStep; }
```

Comment:
Might as well remove these commented out functions. It was sloppy of me to leave them in.